velox/connectors/hive/storage_adapters/abfs/AbfsWriteFile.cpp

```diff
@@ -126,25 +111,35 @@ class AbfsWriteFile::Impl {
   }
 
   const std::string path_;
-  const std::string connectStr_;
-  std::string fileSystem_;
-  std::string fileName_;
-  std::shared_ptr<IBlobStorageFileClient> blobStorageFileClient_;
+  const std::shared_ptr<AzureDataLakeFileClient> client_;
 
   uint64_t position_ = -1;
   bool closed_ = false;
 };
 
 AbfsWriteFile::AbfsWriteFile(
-    const std::string& path,
-    const std::string& connectStr) {
-  impl_ = std::make_shared<Impl>(path, connectStr);
+    std::string_view path,
+    const config::ConfigBase& config) {
+  auto abfsAccount = AbfsConfig(path, config);
+  std::shared_ptr<AzureDataLakeFileClient> client =
+      std::make_shared<DataLakeFileClientWrapper>(
```

Comment:
Just curious why the DataLakeFileClientWrapper is shared_ptr and not unique_ptr? Would something else access this if it is part of the client?

Reply:
shared_ptr made it easier to write tests. But I changed this to a unique_ptr as I think it should be as well.